src/lazy/system_stream_item.rs

```diff
@@ -73,17 +73,16 @@ impl<'top, D: Decoder> SystemStreamItem<'top, D> {
         }
     }
 
-    /// Like [`Self::symbol_table`], but returns a [`IonError::Decoding`] if this item is not
-    /// a symbol table.
-    pub fn symbol_table(self) -> Option<LazyStruct<'top, D>> {
+    /// If this item is a symbol table, returns `Some(lazy_struct)`. Otherwise, returns `None`.
+    pub fn as_symbol_table(self) -> Option<LazyStruct<'top, D>> {
```

Comment:
🗺️ I renamed this method for consistency; other types use `as_something`/`expect_something` for their `Option<T>`/`IonResult<T>` pairs.